Backup collections doc

Enjection opened the pull request:
Related: #22154

### Changelog entry <!-- a user-readable short description of the changes that goes to CHANGELOG.md and Release Notes -->

...

### Changelog category <!-- remove all except one -->

* Documentation (changelog entry is not required)

### Description for reviewers <!-- (optional) description for those who read this PR -->

...

CyberROFL commented on ydb/docs/en/core/reference/ydb-cli/export-import/_includes/index.md:
```diff
@@ -2,6 +2,17 @@
 
 The {{ ydb-short-name }} CLI contains a set of commands designed to export and import data and descriptions of data schema objects. Data can be exported to create backups for subsequent recovery and for other purposes.
 
+## Backup collections (New)
```
А точно стоит помечать это как New? Мы вроде так никогда не делали.

CyberROFL commented on ydb/docs/en/core/reference/ydb-cli/export-import/backup-collections/concepts.md:
```diff
@@ -0,0 +1,102 @@
+
+{% note warning %}
+
+**Critical:** Never delete full backups that have dependent incremental backups. Deleting a full backup breaks the entire chain, making all subsequent incrementals unrestorable.
```
Лучше сделать красную плашку, чем писать в теле "Critical".

CyberROFL commented on ydb/docs/en/core/reference/ydb-cli/export-import/backup-collections/index.md:
```diff
@@ -0,0 +1,30 @@
+# Backup collections
+
+Efficient incremental backup and point-in-time recovery for YDB tables (v25.3+).
```
У нас и так версионированная документация, можно убрать отсюда упоминание версии.

CyberROFL commented on ydb/docs/en/core/reference/ydb-cli/export-import/backup-collections/index.md:
````diff
@@ -0,0 +1,30 @@
+
+## Quick start
+
+```sql
+-- 1. Create collection
+CREATE BACKUP COLLECTION `my_backups`
+    ( TABLE `/Root/db/table1`, TABLE `/Root/db/table2` )
+WITH ( STORAGE = 'cluster', INCREMENTAL_BACKUP_ENABLED = 'true' );
+
+-- 2. Take full backup
+BACKUP `my_backups`;
+
+-- 3. Take incremental backups
+BACKUP `my_backups` INCREMENTAL;
+```
````
Не понял, чем это отличается от core/concepts/backup/collections.md. Как будто одно и то же в разных местах доки.

CyberROFL commented on ydb/docs/en/core/reference/ydb-cli/export-import/backup-collections/sql-api.md:
Описание синтаксиса нужно унести в yql reference.

CyberROFL commented on ydb/docs/en/core/reference/ydb-cli/export-import/backup-collections/operations.md:
```diff
@@ -0,0 +1,232 @@
+ydb scheme ls .backups/collections/shop_backups/
+
+# Remove specific backup directories using rmdir
+ydb scheme rmdir .backups/collections/shop_backups/backup_20240315/
```
Без `-r` не будет работать.

CyberROFL commented on ydb/docs/en/core/reference/ydb-cli/export-import/backup-collections/operations.md:
```diff
@@ -0,0 +1,232 @@
+# Remove specific backup directories using rmdir
+ydb scheme rmdir .backups/collections/shop_backups/backup_20240315/
+
+# For table-based cleanup, use DROP TABLE
+ydb scheme describe .backups/collections/shop_backups/backup_20240315/
```
Написано "use DROP TABLE", а в примере "scheme describe".

CyberROFL commented on ydb/docs/en/core/reference/ydb-cli/export-import/backup-collections/operations.md:
````diff
@@ -0,0 +1,232 @@
+
+```bash
+# List operations (may include backup operations)
+ydb operations list
````
После `list` должно быть упоминание типа операции (incbackup).

CyberROFL commented on ydb/docs/en/core/reference/ydb-cli/export-import/backup-collections/operations.md:
```diff
@@ -0,0 +1,232 @@
+ydb operations get <operation-id>
+
+# List operations with filter
+ydb operations list --filter incbackup
```
Впервые встречаю такой синтаксис. У нас так работает?

CyberROFL reviewed (commented):
А где YQL reference?

pixcc commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,160 @@
+- **New rows**: Added since last backup.
+- **Modified rows**: Changed data in existing rows.  
+- **Deleted rows**: Removed data (tombstone records).
+- **Schema changes**: Currently not supported.
```
Может плашкой какой сделать?

pixcc commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,160 @@
+
+### Benefits
+
+- **Storage efficiency**: Incremental backups use significantly less storage.
```
Относительно чего?

CyberROFL commented:
Появилась идея унести "Резервное копирование и восстановление" в концепции и оттуда погружать в разные виды операций: экспорт-импорт, dump-restore, инкрементальный бэкап, backup collections.

s-sabitova commented on ydb/docs/en/core/concepts/backup.md:
```diff
@@ -0,0 +1,49 @@
+For production workloads requiring incremental backups:
+
+- **Use cases**: Production environments, large datasets, regular backup schedules.
+- **Storage**: Currently supports cluster storage only.
```
Для меня не совсем очевидна предложенная классификация способов бэкапа/рестора. 

- export/import и dump/restore - это сами процессы выгрузки и загрузки данных в базу
- backup collection - это объединенный в директорию набор связанных между собой бэкапов

Эти критерии классификации имхо из разных измерений.

Предлагаю концептуально описать две классификации бэкапов: 

Первая - по типу хранилища, в которое осуществляется бэкап. Это похоже на текущую формулировку, но убирает диссонанс, который я описала в самом начале.

Предлагаю сделать одну сравнительную табличку, в которой будут собраны приведенные в текущем тексте детали, вроде такой:

<img width="1762" height="306" alt="image" src="https://github.com/user-attachments/assets/27791afc-e6ac-4280-864e-c3bd6e6e5310" />

Вторая классификация - по методу резервного копирования

1. Полный снапшот выбранных объектов
2. Инкрементальный бэкап

Пользователи будут ходить в доку и целенаправленно искать разделы и статьи про инкрементальные бэкапы, поэтому такая классификация тоже нужна. 

Предлагаю тоже оформить в таблицу:

<img width="2356" height="318" alt="image" src="https://github.com/user-attachments/assets/86c9e9eb-d0e8-4c16-b1c4-a4cd5bd0c548" />

s-sabitova commented on ydb/docs/en/core/maintenance/manual/backup-collections.md:
````diff
@@ -0,0 +1,287 @@
+WITH ( STORAGE = 'cluster', INCREMENTAL_BACKUP_ENABLED = 'true' );
+```
+
+## Taking backups {#taking-backups}
````
Taking имхо здесь не самое подходящее слово.

Возможно, лучше заменить на "making backups" или "creating backups" или "backing up"?

s-sabitova commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,160 @@
+- **Deleted rows**: Removed data (tombstone records).
+- **Schema changes**: Currently not supported.
+
+## Relationship with incremental backups {#relationship-with-incremental-backups}
```
Этот абзац можно поместить выше в статье, на мой взгляд. И слово relationship тут предлагаю поменять на relation

s-sabitova commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,160 @@
+- Table schemas at backup time. (Implicitly)
+- Data files (full or incremental changes).
+
+### Storage backends {#storage-backends}
```
Имхо, точнее Storage options, но не настаиваю

s-sabitova commented on ydb/docs/en/core/maintenance/manual/backup-collections.md:
```diff
@@ -0,0 +1,287 @@
+
+### Initial full backup {#initial-full-backup}
+
+After creating a collection, take the initial full backup:
```
Аналогично - заменить take на make?

s-sabitova commented on ydb/docs/en/core/maintenance/manual/backup-collections.md:
````diff
@@ -0,0 +1,287 @@
+    ( TABLE `/Root/shop/orders`, TABLE `/Root/shop/products` )
+WITH ( STORAGE = 'cluster', INCREMENTAL_BACKUP_ENABLED = 'true' );
+```
+
````
Предлагаю явно указать, какие значения STORAGE и INCREMENTAL_BACKUP_ENABLED можно использовать.

s-sabitova commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,160 @@
+1. **Collection creation**: Define which tables to include and storage settings
+2. **Initial full backup**: Create baseline snapshot of all tables
+3. **Regular incremental backups**: Capture ongoing changes on-demand
+4. **Chain management**: Monitor backup chains and manage retention manually
```
Имхо не хватает пояснения, как именно это (monitoring & managing retention) делать. Если такое описание есть, предлагаю здесь добавить ссылку на него

s-sabitova commented on ydb/docs/en/core/maintenance/manual/backup-collections.md:
```diff
@@ -0,0 +1,287 @@
+ydb operation list incbackup
+
+# Get details for specific operation
+ydb operation get <operation-id>
```
Немного путает, что под <operation-id> подразумевается полная строка вида "ydb://incbackup/11?id=281474992198178"

Но при этом в ней есть параметр id

Можно подумать, что команда должна выглядеть так ydb operation get 281474992198178

Предлагаю вместо <operation-id> тут написать что-то другое

CyberROFL commented on ydb/docs/ru/core/yql/reference/syntax/update.md:
Лишний дифф.

CyberROFL commented on ydb/docs/ru/core/yql/reference/syntax/backup.md:
````diff
@@ -0,0 +1,84 @@
+Полная резервная копия создает полный снимок всех таблиц в коллекции на определенный момент времени. Это служит базовой линией для последующих инкрементальных резервных копий.
+
+```yql
+BACKUP `production_backups`;
````
```suggestion
BACKUP production_backups;
```

CyberROFL commented on ydb/docs/ru/core/yql/reference/syntax/backup.md:
```diff
@@ -0,0 +1,84 @@
+
+### Полная резервная копия
+
+Полная резервная копия создает полный снимок всех таблиц в коллекции на определенный момент времени. Это служит базовой линией для последующих инкрементальных резервных копий.
```
```suggestion
Полная резервная копия создает снимок всех таблиц в коллекции на определенный момент времени. Это служит основой для последующих инкрементальных резервных копий.
```

CyberROFL commented on ydb/docs/ru/core/yql/reference/syntax/backup.md:
````diff
@@ -0,0 +1,84 @@
+Инкрементальная резервная копия захватывает только изменения (вставки, обновления, удаления) с момента предыдущей резервной копии в цепочке. Коллекция должна быть создана с `INCREMENTAL_BACKUP_ENABLED = 'true'`.
+
+```yql
+BACKUP `production_backups` INCREMENTAL;
````
```suggestion
BACKUP production_backups INCREMENTAL;
```

CyberROFL commented on ydb/docs/ru/core/yql/reference/syntax/create-backup-collection.md:
```diff
@@ -0,0 +1,58 @@
+# CREATE BACKUP COLLECTION
+
+Выражение `CREATE BACKUP COLLECTION` создает [коллекцию резервных копий](../../../concepts/backup-collections.md) - именованный набор скоординированных резервных копий для выбранных объектов базы данных.
```
```suggestion
Выражение `CREATE BACKUP COLLECTION` создает [коллекцию резервных копий](../../../concepts/backup-collections.md).
```

CyberROFL commented on ydb/docs/ru/core/yql/reference/syntax/create-backup-collection.md:
```diff
@@ -0,0 +1,58 @@
+) WITH (
+    STORAGE = 'cluster',
+    INCREMENTAL_BACKUP_ENABLED = 'true'
+);
```
```suggestion
CREATE BACKUP COLLECTION production_backups (
    TABLE orders,
    TABLE products,
    TABLE customers
) WITH (
    STORAGE = 'cluster',
    INCREMENTAL_BACKUP_ENABLED = 'true'
);
```

CyberROFL commented on ydb/docs/ru/core/yql/reference/syntax/drop-backup-collection.md:
````diff
@@ -0,0 +1,32 @@
+Выражение `DROP BACKUP COLLECTION` удаляет [коллекцию резервных копий](../../../concepts/backup-collections.md) и все содержащиеся в ней резервные копии.
+
+```yql
+DROP BACKUP COLLECTION `имя_коллекции`;
````
```suggestion
DROP BACKUP COLLECTION collection_name;
```

CyberROFL commented on ydb/docs/ru/core/yql/reference/syntax/restore-backup-collection.md:
```diff
@@ -0,0 +1,55 @@
+* Восстанавливает все таблицы из последней резервной копии в коллекции
+* Если существуют инкрементальные резервные копии, применяет их поверх последней полной резервной копии
+* Восстанавливает таблицы в их состояние на момент резервной копии
+* Перезаписывает существующие таблицы с теми же именами
```
```suggestion
* Восстанавливает все таблицы из последней резервной копии в коллекции.
* Если существуют инкрементальные резервные копии, применяет их поверх последней полной резервной копии.
* Восстанавливает таблицы в их состояние на момент резервной копии.
* Перезаписывает существующие таблицы с теми же именами.
```

CyberROFL commented on ydb/docs/ru/core/yql/reference/syntax/restore-backup-collection.md:
```diff
@@ -0,0 +1,55 @@
+
+## Мониторинг операций восстановления
+
+Операции восстановления выполняются асинхронно в фоновом режиме. Вы можете отслеживать их прогресс с помощью YDB CLI:
```
```suggestion
Операции восстановления выполняются асинхронно в фоновом режиме. Вы можете отслеживать их прогресс с помощью {{ ydb-short-name }} CLI:
```

CyberROFL commented on ydb/docs/ru/core/yql/reference/syntax/restore-backup-collection.md:
```diff
@@ -0,0 +1,55 @@
+* [CREATE BACKUP COLLECTION](create-backup-collection.md)
+* [BACKUP](backup.md)
+* [DROP BACKUP COLLECTION](drop-backup-collection.md)
+* [Руководство по операциям](../../../maintenance/manual/backup-collections.md)
```
```suggestion
* [Коллекции резервных копий](../../../concepts/backup-collections.md).
* [CREATE BACKUP COLLECTION](create-backup-collection.md).
* [BACKUP](backup.md).
* [DROP BACKUP COLLECTION](drop-backup-collection.md).
* [Руководство по операциям с коллекциями резервных копий](../../../maintenance/manual/backup-collections.md).
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup.md:
```diff
@@ -0,0 +1,43 @@
+
+Резервные копии могут храниться в различных местах в зависимости от ваших требований:
+
+|| Тип хранилища | Тип резервной копии | Лучше всего для | Как использовать |
```
```suggestion
|| Тип хранилища | Тип резервной копии | Сценарий использования | Как использовать |
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup.md:
```diff
@@ -0,0 +1,43 @@
+|| Тип хранилища | Лучше всего для | Как использовать | Ограничения |
+||---------------|-----------------|------------------|-------------|
+|| **Полный снимок** | Локально в базе данных, s3-совместимое хранилище, файловая система | Разовые резервные копии, миграция базы данных, тестирование, небольшие производственные среды | ydb export/import или ydb dump/restore или ссылка на [документацию CLI](../reference/ydb-cli/export-import/index.md) | - |
+|| **Инкрементальное резервное копирование** | Локально в базе данных, s3-совместимое хранилище*, файловая система* | Производственные среды, большие наборы данных, регулярные расписания резервного копирования | Требуются коллекции резервных копий - ссылка на [документацию коллекций](backup-collections.md) | Пока резервируются только таблицы |
```
```suggestion
|| **Инкрементальное резервное копирование** | Локально в базе данных, s3-совместимое хранилище\*, файловая система\* | Производственные среды, большие наборы данных, регулярные расписания резервного копирования | Требуются [коллекции резервных копий](backup-collections.md) | Поддерживаются только строковые таблицы |
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup.md:
```diff
@@ -0,0 +1,43 @@
+
+- [Концепции коллекций резервных копий](backup-collections.md) - Архитектура и концепции.
+- [Руководство по операциям](../maintenance/manual/backup-collections.md) - Практические операции.
+- [Общие рецепты](../recipes/backup-collections.md) - Примеры использования.
```
```suggestion
- [Коллекции резервных копий](backup-collections.md).
- [Руководство по операциям с коллекциями резервных копий](../maintenance/manual/backup-collections.md).
- [Примеры использования коллекций резервных копий](../recipes/backup-collections.md).
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,304 @@
+# Коллекции резервных копий
+
+Коллекции резервных копий обеспечивают продвинутое решение для резервного копирования в YDB, которое организует полные и инкрементальные резервные копии в управляемые коллекции. Этот подход предназначен для производственных нагрузок, требующих эффективного аварийного восстановления и возможностей восстановления на момент времени.
```
```suggestion
Коллекции резервных копий обеспечивают продвинутое решение для резервного копирования в {{ ydb-short-name }}, которое организует полные и инкрементальные резервные копии в управляемые коллекции. Этот подход предназначен для производственных нагрузок, требующих эффективного аварийного восстановления и возможностей восстановления на момент времени.
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,304 @@
+
+### Цепочка резервных копий {#backup-chain}
+
+Упорядоченная последовательность резервных копий, начинающаяся с полной резервной копии, за которой следует ноль или более инкрементальных резервных копий. Каждая инкрементальная резервная копия зависит от всех предыдущих резервных копий в цепочке для полного восстановления.
```
```suggestion
Упорядоченная последовательность резервных копий, начинающаяся с полной резервной копии, за которой следует ноль или более инкрементальных резервных копий. Каждая инкрементальная резервная копия зависит от всех предыдущих резервных копий в цепочке.
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,304 @@
+
+Создает именованный контейнер, который координирует резервные копии нескольких объектов базы данных. Все объекты в коллекции используют одинаковую конфигурацию резервного копирования и место хранения. После создания список объектов фиксируется и не может быть изменен.
+
+[См. руководство по операциям](../maintenance/manual/backup-collections.md#create-collection)
```
```suggestion
[См. руководство по операциям](../maintenance/manual/backup-collections.md#create-collection).
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,304 @@
+
+Создает транзакционно согласованный снимок всех объектов в коллекции в определенный момент времени. Эта базовая резервная копия содержит все данные и схемы и может быть восстановлена независимо без необходимости в других резервных копиях. Полная резервная копия требуется перед созданием любых инкрементальных резервных копий.
+
+[См. руководство по операциям](../maintenance/manual/backup-collections.md#create-backup)
```
```suggestion
[См. руководство по операциям](../maintenance/manual/backup-collections.md#create-backup).
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,304 @@
+
+Захватывает только данные, которые изменились с момента предыдущей резервной копии, что значительно снижает требования к хранилищу. **Важно**: Встроенный планировщик отсутствует, поэтому вы должны запускать инкрементальные резервные копии вручную или использовать внешнюю систему планирования. Для восстановления из инкрементальной резервной копии вам нужна вся цепочка резервных копий, начиная с на ...
+
+[См. руководство по операциям](../maintenance/manual/backup-collections.md#create-backup)
```
```suggestion
[См. руководство по операциям](../maintenance/manual/backup-collections.md#create-backup).
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,304 @@
+
+Если ваши резервные копии хранятся во внешнем хранилище (файловой системе или объектном хранилище), импортируйте их обратно в кластер с помощью операций импорта YDB CLI. Импортируйте все резервные копии в цепочке в правильном порядке, начиная с полной резервной копии, за которой следуют инкрементальные резервные копии. Этот шаг необходим только при восстановлении из внешнего хранилища.
+
+[См. экспорт/импорт YDB CLI](../reference/ydb-cli/export-import/index.md)
```
```suggestion
[См. экспорт/импорт YDB CLI](../reference/ydb-cli/export-import/index.md).
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,304 @@
+
+Создайте новую коллекцию, которая определяет, куда будут помещены восстановленные данные. Структура коллекции должна соответствовать исходной коллекции резервных копий, включая те же имена таблиц и пути. Это подготавливает целевое расположение для процесса восстановления.
+
+[См. руководство по операциям](../maintenance/manual/backup-collections.md#restore-backup)
```
```suggestion
[См. руководство по операциям](../maintenance/manual/backup-collections.md#restore-backup).
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,304 @@
+
+После успешного восстановления и проверки ваших данных удалите коллекцию восстановления, чтобы очистить метаданные и освободить ресурсы. Это необязательный шаг очистки, который помогает поддерживать порядок в каталоге резервных копий.
+
+[См. руководство по операциям](../maintenance/manual/backup-collections.md#cleanup)
```
```suggestion
[См. руководство по операциям](../maintenance/manual/backup-collections.md#cleanup).
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,304 @@
+
+[См. руководство по операциям](../maintenance/manual/backup-collections.md#cleanup)
+
+## Операционное руководство {#operational-guidance}
```
Какое-то странное название.

CyberROFL reviewed (changes requested):
Прочитал не всё, но пока есть что поправить. В основном можно применять правки и двигаться дальше.

CyberROFL reviewed (dismissed):
Прочитал не всё, но пока есть что поправить. В основном можно применять правки и двигаться дальше.

s-sabitova commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,306 @@
+
+### Backup collection {#backup-collection}
+
+A named container that groups backups for a specific set of database objects (currently tables only). Collections ensure that all included objects are backed up consistently.
```
тут друг за другом приводится два определения бэкап коллекции:

A backup collection is a named set of coordinated backups for selected database objects.

A named container that groups backups for a specific set of database objects (currently tables only).

может, оставить что-то одно?

s-sabitova commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,306 @@
+
+[See YDB CLI export/import](../reference/ydb-cli/export-import/index.md)
+
+### Create collection for restore {#restore-collection-creation}
```
насколько я понимаю, сначала должен идти шаг с созданием коллекции, а затем - с импортом в нее данных из внешнего хранилища?

s-sabitova commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,306 @@
+
+[See YDB CLI export/import](../reference/ydb-cli/export-import/index.md) | [See monitoring guide](../maintenance/manual/backup-collections.md#monitoring) | [See cleanup procedures](../maintenance/manual/backup-collections.md#cleanup)
+
+## Architecture and components {#architecture}
```
Я бы этот раздел переместила повыше: после глоссария, где вводятся основные понятия, но до описания backup flow

s-sabitova commented on ydb/docs/en/core/concepts/backup.md:
```diff
@@ -0,0 +1,43 @@
+
+\* In order to store incremental backup chain in s3-compatible storage or file system it is required to make a backup collection locally first, and then to back it up its directory using full snapshot backup.
+
+## Backup collections {#backup-collections}
```
мне кажется, достаточно тут сослаться на статью про бэкап коллекции (это сделано выше), не нужно повторять тут определение и детали про экспорты-импорты

s-sabitova commented on ydb/docs/en/core/recipes/backup-collections.md:
```diff
@@ -0,0 +1,199 @@
+# Backup Collections: Common Recipes and Examples
```
может быть, слить этот файл и ydb/docs/en/core/maintenance/manual/backup-collections.md ? мне кажется, может получиться дублирование, что может запутать

CyberROFL commented on ydb/docs/ru/core/yql/reference/syntax/restore-backup-collection.md:
```diff
@@ -0,0 +1,55 @@
+* Восстанавливает все таблицы из последней резервной копии в коллекции
+* Если существуют инкрементальные резервные копии, применяет их поверх последней полной резервной копии
+* Восстанавливает таблицы в их состояние на момент резервной копии
+* Перезаписывает существующие таблицы с теми же именами
```
```suggestion
* Восстанавливает все таблицы из последней резервной копии в коллекции.
* Если существуют инкрементальные резервные копии, применяет их поверх последней полной резервной копии.
* Восстанавливает таблицы в их состояние на момент резервной копии.
* Перезаписывает существующие таблицы с теми же именами.
```

CyberROFL commented on ydb/docs/ru/core/recipes/backup-collections.md:
````diff
@@ -0,0 +1,254 @@
+
+```sql
+-- Создание коллекции для связанных таблиц
+CREATE BACKUP COLLECTION `production_backups`
````
Можно в этом файле все примеры сделать без бэктиков и полных путей?

CyberROFL commented on ydb/docs/ru/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,300 @@
+
+1. **Изоляция транзакций**: Резервное копирование начинается с согласованной точки снимка.
+2. **Отслеживание изменений**: Для инкрементальных резервных копий захватываются только изменения с последней резервной копии и сохраняются в [changefeed](cdc.md).
+3. **Материализация изменений**: При вызове инкрементального резервного копирования changefeed компактируется в таблицы инкрементальной резервной копии.
```
```suggestion
3. **Материализация изменений**: При вызове инкрементального резервного копирования поток изменений компактифицируется в таблицы инкрементальной резервной копии.
```

blinkov commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,308 @@
+# Backup collections
```
From neighboring articles, it sounds like backup collections are a schema object kind, thus this article belongs to the https://ydb.tech/docs/en/concepts/datamodel/?version=v25.2 folder.

blinkov replied:
Add "backup collections" entry to glossary.md with a brief description and a link to this article for further details.

blinkov commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,308 @@
+# Backup collections
+
+Backup collections provide an advanced backup solution for {{ ydb-short-name }} that organizes full and incremental backups into managed collections. This approach is designed for production workloads requiring efficient disaster recovery and point-in-time recovery capabilities.
```
The second sentence doesn't explain how these claims are achieved. If this is indeed a schema object, give a high-level explanation of how it helps recovery if the whole cluster is gone for whatever reason. Or is the main use case recovery of like individual tables removed due to human error? Again, state this in the beginning so it's clear what kind of backups we're talking about and what to expect.

blinkov commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,308 @@
+
+An ordered sequence of backups starting with a full backup followed by zero or more incremental backups. Each incremental backup depends on all previous backups in the chain.
+
+## Backup workflow {#backup-workflow}
```
Add an intro paragraph with high-level explanation of what's going on in the following diagram

blinkov commented on ydb/docs/en/core/maintenance/manual/backup-collections.md:
```diff
@@ -0,0 +1,311 @@
+# Backup Collections Operations
```
`maintenance/` is a deprecated folder in favor of `devops/` — new content shouldn't be placed here.

blinkov commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,308 @@
+
+### Incremental backups {#incremental-backups-creation}
+
+Captures only data that changed since the previous backup, significantly reducing storage requirements. **Important**: Incremental backups must be triggered manually or using an external scheduling system. To restore from an incremental backup, you need the entire chain of backups starting from the initial full backup.
```
use `{% note %}` for important stuff, not just bold

blinkov replied:
Explain what exactly "changed" means in this context. At least, whether we're backing up all modifications happened since the previous backup or just somehow deduplicating rows.

blinkov commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,308 @@
+
+### Initial full backup {#initial-full-backup}
+
+Creates a consistent snapshot of all objects in the collection at a specific point in time. This baseline backup contains all data and can be restored independently without requiring other backups. A full backup is required before creating any incremental backups.
```
Explain what guarantees to expect from this. For instance, is it executed against a global snapshot that ensures referential integrity across objects?

blinkov commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,308 @@
+
+### Incremental backup mechanism {#incremental-backup-mechanism}
+
+{% note info %}
```
It looks odd to wrap the whole section that has a header into a `{% note %}`

blinkov commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,308 @@
+- **Chain management**: Collections manage the sequence of full and incremental backups.
+- **Consistency**: All tables in a collection are backed up consistently.
+
+Without backup collections, only full export/import operations are available.
```
This doesn't seem to add any new information beyond what has been covered above.

blinkov commented on ydb/docs/en/core/concepts/backup.md:
```diff
@@ -0,0 +1,43 @@
+|| Backup type | Storage type | Use cases | How to use | Limitations |
+||-------------|--------------|-----------|------------|-------------|
+|| **Full snapshot** | Locally in database, s3-compatible storage, file system | One-time backups, database migration, testing, smaller production environments | ydb export/import or ydb dump/restore or link to [CLI docs](../reference/ydb-cli/export-import/index.md) | - |
+|| **Incremental backup** | Locally in database, s3-compatible storage\*, file system\* | Production environments, large datasets, regular backup schedules | Requires [backup collections](backup-collections.md) | Only row-oriented tables supported |
```
`s/s3/S3/g`

blinkov commented on ydb/docs/en/core/concepts/backup.md:
```diff
@@ -0,0 +1,43 @@
+## See also
+
+- [Backup and recovery guide](../devops/backup-and-recovery.md).
+- [Export and import reference](../reference/ydb-cli/export-import/index.md).
```
Merge these two lists + add links to relevant pages in YQL reference

blinkov commented on ydb/docs/en/core/maintenance/manual/backup-collections.md:
```diff
@@ -0,0 +1,311 @@
+### Important scheduling notes {#important-scheduling-notes}
+
+- **External scheduling required**: YDB does not provide built-in scheduling. Use cron or similar tools.
+- **Background operations**: Backups run asynchronously and don't block database operations.
```
But they probably do consume some resources while they happen? How to avoid overruning the cluster with too many backups happening concurrently?

blinkov commented on ydb/docs/en/core/maintenance/manual/backup-collections.md:
````diff
@@ -0,0 +1,311 @@
+After creating a collection, take the initial full backup:
+
+```sql
+BACKUP `shop_backups`;
````
Clarify whether these queries are idempotent or not, is it safe to retry many times if you're getting timeouts or something similar?

blinkov commented on ydb/docs/en/core/maintenance/manual/backup-collections.md:
```diff
@@ -0,0 +1,311 @@
+
+### Collection information {#collection-information}
+
+Since collections are managed through SQL, browse them using schema commands:
```
This sounds counterintuitive: we're managing them with SQL, but navigating with CLI — why?

Enjection replied:
because our SQL doesn't allow to navigate schema

blinkov commented on ydb/docs/en/core/maintenance/manual/backup-collections.md:
```diff
@@ -0,0 +1,311 @@
+1. **Verify recent backups exist**.
+2. **Check backup chain completeness**.
+3. **Monitor storage usage**.
+4. **Validate backup accessibility**.
```
Provide specific instructions on how exactly to do that. Also, give recommendations that you should regularly check that the backups are restoring correctly with some sanity checks, again with examples/instructions.

blinkov commented on ydb/docs/en/core/maintenance/manual/backup-collections.md:
```diff
@@ -0,0 +1,311 @@
+# 4. Remove old backup directories (entire chains only)
+ydb scheme rmdir -r .backups/collections/shop_backups/20250208141425Z_full/
+ydb scheme rmdir -r .backups/collections/shop_backups/20250209141519Z_incremental/
+ydb scheme rmdir -r .backups/collections/shop_backups/20250210141612Z_incremental/
```
Explain how to understand that these incrementals belong to `20250208141425Z_full`? Just because they are following each other chronologically?

blinkov commented on ydb/docs/en/core/recipes/backup-collections.md:
```diff
@@ -0,0 +1,194 @@
+# Recipes and examples
```
Too generic header. Refactor this into a recipe folder and put each scenario into a separate file with a verbose, fully matching title for SEO purposes. Link to all scenarios from index.md of the new folder and from relevant parts of other articles above.

blinkov commented on ydb/docs/en/core/maintenance/manual/backup-collections.md:
```diff
@@ -0,0 +1,311 @@
+- **Performance tracking**: Monitor backup duration and resource usage.
+- **Storage monitoring**: Track backup storage growth.
+
+## See also {#see-also}
```
Link to all new articles, not just these two

blinkov commented on ydb/docs/en/core/maintenance/manual/backup-collections.md:
````diff
@@ -0,0 +1,311 @@
+
+```bash
+# Export backup collection from source
+ydb tools dump -p .backups/collections/shop_backups -o shop_backups_export
````
What to do if the backup is too large to be exported locally?

blinkov commented on ydb/docs/en/core/maintenance/manual/backup-collections.md:
```diff
@@ -0,0 +1,311 @@
+# WITH ( STORAGE = 'cluster', INCREMENTAL_BACKUP_ENABLED = 'true' );
+
+# Import to target database
+ydb tools restore -i shop_backups_export -d /Root/restored_db
```
What to do if you only have one database? How to restore to a different folder or different filenames?

blinkov commented on ydb/docs/en/core/reference/ydb-cli/operation-list.md:
```diff
@@ -13,8 +13,12 @@ General format of the command:
   * `buildindex`: The build index operations.
-  * `export/s3`: The export operations.
-  * `import/s3`: The import operations.
+  * `export/yt`: The export to YT operations.
```
what YT and `scriptexec` have to do with the topic of this PR?

blinkov commented on ydb/docs/en/core/yql/reference/syntax/backup.md:
````diff
@@ -0,0 +1,80 @@
+The `BACKUP` statement creates a backup of tables in a [backup collection](../../../concepts/backup-collections.md).
+
+```yql
+BACKUP collection_name [INCREMENTAL];
````
You indeed can't write `FULL` if you don't want to rely on the default?

Enjection commented on ydb/docs/en/core/concepts/backup-collections.md:
```diff
@@ -0,0 +1,308 @@
+
+### Chain management {#chain-management-details}
+
+You are responsible for tracking and maintaining your backup chains. Use `ydb scheme ls` to view the chain structure and identify which backups can be safely deleted. Be careful when removing backups to avoid breaking chain dependencies, as incremental backups rely on their parent backups being available.
```
We will have them at some point, not implemented yet due to lack of time/

Enjection commented on ydb/docs/en/core/maintenance/manual/backup-collections.md:
```diff
@@ -0,0 +1,311 @@
+After backup completion:
+
+1. **Verify operation success**: Check operation status.
+2. **Validate backup structure**: Browse backup directories.
```
It's quite complex. As far as we expose the guts of our backups..

CyberROFL commented on ydb/docs/ru/core/yql/reference/syntax/create-backup-collection.md:
```diff
@@ -0,0 +1,57 @@
+
+{% note info %}
+
+При выборе имени для коллекции резервных копий учитывайте общие [правила именования объектов схемы](../../../concepts/datamodel/cluster-namespace.md#object-naming-rules).
```
Как будто лишнее. Не помню, чтобы мы в других местах делали такую плашку.

AndreyVepritskiyQA commented on ydb/docs/en/core/recipes/backup-collections/importing-and-restoring.md:
````diff
@@ -0,0 +1,54 @@
+
+```bash
+# Step 1: Import full backup
+ydb tools restore -i /backup/full_20250821 -d .backups/collections/recovery/20250821141425Z_full
````
Импорт одного бэкапа коллекции из файлового хранилища

**Ожидаемый результат:**
Бэкап перенесен из файлового хранилища в базу данных

**Фактические результат:**
Выполнение команды падает с ошибкой `unknown option 'd' in '-d'`


**Дополнительно:**
Получилось выполнить команду со следующим синтаксисом: 
```
ydb tools restore -p ".backups/collections/production_backup/20260331111331Z_full" -i "backup/
20260331111331Z_full"
```

AndreyVepritskiyQA commented on ydb/docs/ru/core/recipes/backup-collections/importing-and-restoring.md:
````diff
@@ -0,0 +1,54 @@
+
+```bash
+# Шаг 1: Импорт полной резервной копии
+ydb tools restore -i /backup/full_20250821 -d .backups/collections/recovery/20250821141425Z_full
````
**Идентичная проблема из EN документации**

Импорт одного бэкапа коллекции из файлового хранилища

**Ожидаемый результат:**
Бэкап перенесен из файлового хранилища в базу данных

**Фактические результат:**
Выполнение команды падает с ошибкой `unknown option 'd' in '-d'`


**Дополнительно:**
Получилось выполнить команду со следующим синтаксисом: 
```
ydb tools restore -p ".backups/collections/production_backup/20260331111331Z_full" -i "backup/
20260331111331Z_full"
```

AndreyVepritskiyQA commented on ydb/docs/ru/core/recipes/backup-collections/validation-and-testing.md:
```diff
@@ -0,0 +1,110 @@
+ydb export s3 \
+  --s3-endpoint storage.yandexcloud.net \
+  --bucket my-backup-bucket \
+  --item .backups/collections/production_backups/20250821141425Z_full,backups/test_export
```
**Идентичная проблема из EN документации**

Экспорт коллекции бэкапов в хранилище S3

**Ожидаемый результат:**
Файлы бэкапов перенесены в хранилище S3

**Фактический результат:**
Ошибка при выполнении команды экспорта

<img width="736" height="143" alt="Image" src="https://github.com/user-attachments/assets/40e783c7-631a-4d67-be52-cfe2aa3d7fee" />

**Дополнительно:**
Рабочий вариант команды был представлен в разделе https://github.com/ydb-platform/ydb/pull/23021/changes#diff-e799e65c3266654ac2f717068bb7f6f489813124b8ad80ac9ba7178b8e287e06R204

Рабочий вариант команды:

```bash
ydb export s3 --s3-endpoint storage.example.com \
  --bucket my-backups \
  --item src=.backups/collections/production_backups,dst=production_backups
```

AndreyVepritskiyQA commented on ydb/docs/ru/core/recipes/backup-collections/validation-and-testing.md:
```diff
@@ -0,0 +1,110 @@
+ydb tools dump -p .backups/collections/production_backups/backup_20240315_120000 -o /tmp/test_restore
+
+# Тестовое восстановление во временное место (в тестовой среде)
+ydb tools restore -i /tmp/test_restore -d /Root/test_restore_verification
```
**Идентичная проблема из EN документации**

Импорт одного бэкапа коллекции из файлового хранилища

**Ожидаемый результат:**
Бэкап перенесен из файлового хранилища в базу данных

**Фактические результат:**
Выполнение команды падает с ошибкой `unknown option 'd' in '-d'`

**Дополнительно:**
Получилось выполнить команду со следующим синтаксисом:

`ydb tools restore -p ".backups/collections/test_restore_verification" -i "/tmp/test_restore"`

CyberROFL reviewed (commented):
1. https://github.com/ydb-platform/ydb/pull/23021#discussion_r2576290979
2. https://github.com/ydb-platform/ydb/pull/23021#pullrequestreview-4038935016
3. https://github.com/ydb-platform/ydb/pull/23021#pullrequestreview-4041956834
4. https://github.com/ydb-platform/ydb/pull/23021#pullrequestreview-4042473990
5. https://github.com/ydb-platform/ydb/pull/23021#pullrequestreview-4049119901

Нужно закрыть и перепризвать.

fomichev3000 commented on ydb/docs/ru/core/concepts/backup.md:
```diff
@@ -0,0 +1,55 @@
+|| **Тип хранилища** | **Тип резервной копии** | **Защита от отказа кластера** | **Сценарии использования** | **Как использовать** ||
+|| S3-совместимое хранилище | Полные резервные копии | Да | Крупная миграция данных, архивирование, передача производственных данных | [`ydb export s3`](../reference/ydb-cli/export-import/export-s3.md) / [`ydb import s3`](../reference/ydb-cli/export-import/import-s3.md) ||
+|| Файловая система | Полные резервные копии | Да | Локальная разработка, тестирование, небольшие производственные среды | [`ydb tools dump`](../reference/ydb-cli/export-import/tools-dump.md) / [`ydb tools restore`](../reference/ydb-cli/export-import/tools-restore.md) ||
+|| Кластерное хранилище | Полные и инкрементальные резервные копии | Нет | Производственные среды с регулярным расписанием резервного копирования, большие наборы данных | [Коллекции резервных копий](datamodel/backup-collection.md) ||
```
"Нет" в этой строчке меня убивает! Зачем тогда вообще нужны backup Collections?

Скорее "Да, при условии настройки экспорта"? или что-то подобное

fomichev3000 commented on ydb/docs/ru/core/concepts/backup.md:
```diff
@@ -0,0 +1,55 @@
+|| Инкрементальная резервная копия | Кластер (с возможностью экспорта в S3/файловую систему) | Регулярное резервное копирование, большие наборы данных с низкой частотой изменений | Только [строковые таблицы](datamodel/table.md#row-oriented-tables) ||
+|#
+
+Инкрементальные резервные копии эффективнее по использованию хранилища, чем множество полных резервных копий, так как захватывают только изменения с момента предыдущей резервной копии.
```
Мне кажется эту мысль надо как-то более явно и просто раскрыть. Что вот есть full backup, есть инкременты к нему.

fomichev3000 commented on ydb/docs/ru/core/concepts/backup.md:
```diff
@@ -0,0 +1,55 @@
+
+Инкрементальные резервные копии эффективнее по использованию хранилища, чем множество полных резервных копий, так как захватывают только изменения с момента предыдущей резервной копии.
+
+## Коллекции резервных копий {#backup-collections}
```
Мне не нравится документ. Как будто идея хорошая разделить все на концепции. Но вместо этого акцент идет на backup Collections вместо того, чтобы начать от простого к сложному. Обязательно про снимок и copy table нужно.

fomichev3000 commented on ydb/docs/ru/core/concepts/backup.md:
```diff
@@ -0,0 +1,55 @@
+
+## Коллекции резервных копий {#backup-collections}
+
+[Коллекции резервных копий](datamodel/backup-collection.md) — это [схемные объекты](datamodel/index.md), которые организуют полные и инкрементальные резервные копии структурированным образом и позволяют восстанавливать данные **к состоянию на момент одной из резервных инкрементальных копий в цепочке** (см. [`RESTORE`](../yql/reference/syntax/restore-backup-collection.md)).
```
Странно делать акцент на схемных объектах описывая принцип. Кажется, надо сначала рассказать понятно про принцип, а потом уже про имплементацию в виде схемных объектов.

Меня не покидает ощущение, что информация до читателя доносится слишком сложно и предвзято. Грубо говоря про backup Collections автор хочет рассказать, а про бэкапы -- нет.

А на самом деле пользователя вообще не особо волнуют схемные объекты backup Collections --  это просто способ имплементации

fomichev3000 reviewed (changes requested):
Тяжело читается, хотелось бы реструктурировать так, чтобы дальше можно было конкретные правки предлагать...

CyberROFL commented on ydb/docs/ru/core/reference/ydb-cli/operation-list.md:
```diff
@@ -11,15 +11,11 @@
-  * `scriptexec` — операции выполнения скриптов;
-  * `incbackup` — операции инкрементального резервного копирования;
-  * `restore` — операции восстановления из резервной копии.
+  * `buildindex` — операции построения индекса.
```
Что-то тут список прохудился. Артефакты неудачного разрешения конфликтов?

CyberROFL replied:
Некоторых типов по-прежнему не хватает.

CyberROFL commented on ydb/docs/ru/core/concepts/backup.md:
```diff
@@ -0,0 +1,105 @@
+Подходит для:
+
+- быстрого создания «страховочной» копии перед опасной операцией
+- клонирования данных для тестирования
```
```suggestion
- быстрого создания «страховочной» копии перед опасной операцией;
- клонирования данных для тестирования.
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup.md:
```diff
@@ -0,0 +1,105 @@
+
+- локальной разработки и тестирования
+- небольших баз данных
+- создания копии на отдельном носителе
```
```suggestion
- локальной разработки и тестирования;
- небольших баз данных;
- создания копии на отдельном носителе.
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup.md:
```diff
@@ -0,0 +1,105 @@
+
+- аварийного восстановления (данные хранятся вне кластера)
+- миграции данных между кластерами
+- долгосрочного архивирования
```
```suggestion
- аварийного восстановления (данные хранятся вне кластера);
- миграции данных между кластерами;
- долгосрочного архивирования.
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup.md:
```diff
@@ -0,0 +1,105 @@
+
+{% endnote %}
+
+## Коллекции резервных копий {#backup-collections}
+
+Коллекция резервных копий — это механизм, который объединяет набор таблиц и управляет цепочкой их полных и инкрементальных резервных копий.
+
+### Как это работает
+
+1. **Создание коллекции**: Вы определяете, какие таблицы включить, с помощью [`CREATE BACKUP COLLECTION`](../yql/reference/syntax/create-backup-collection.md).
+
+2. **Полная резервная копия**: Команда [`BACKUP`](../yql/reference/syntax/backup.md) создаёт согласованный снимок всех таблиц в коллекции. Если в коллекции включены инкрементальные копии, одновременно создаются [потоки изменений](cdc.md) на каждой таблице для отслеживания последующих модификаций.
+
+3. **Инкрементальная резервная копия**: Команда `BACKUP ... INCREMENTAL` материализует накопленные в потоках изменений данные в дельта-таблицы. После этого потоки сбрасываются и начинают копить изменения для следующего инкремента.
+
+4. **Восстановление**: Команда [`RESTORE`](../yql/reference/syntax/restore-backup-collection.md) применяет полную копию и все инкременты в цепочке, восстанавливая таблицы к состоянию на момент последней резервной копии.
+
+5. **Удаление**: Команда [`DROP BACKUP COLLECTION`](../yql/reference/syntax/drop-backup-collection.md) удаляет коллекцию и все связанные данные.
+
+### Хранение {#backup-collections-storage}
+
+Коллекции резервных копий хранятся внутри кластера. Как и [копирование таблиц](#copy-table), они защищают от логических ошибок, но не от потери кластера. Для аварийного восстановления регулярно экспортируйте содержимое коллекций во внешнее хранилище с помощью [`{{ ydb-cli }} export s3`](../reference/ydb-cli/export-import/export-s3.md) или [`{{ ydb-cli }} tools dump`](../reference/ydb-cli/export-i ...
+
+Подробнее об архитектуре, ограничениях и внутреннем устройстве коллекций см. в разделе [Коллекции резервных копий](datamodel/backup-collection.md).
```
Я бы это всё унес в стать про коллекции.

CyberROFL replied:
Так и осталось тут.

CyberROFL commented on ydb/docs/ru/core/concepts/backup.md:
```diff
@@ -0,0 +1,105 @@
+|| [`tools copy`](../reference/ydb-cli/tools-copy.md) | В кластере | Нет | Быстрая копия перед опасной операцией ||
+|| [`tools dump`](../reference/ydb-cli/export-import/tools-dump.md) | Файловая система | Нет | Разработка, тестирование, небольшие БД ||
+|| [`export s3`](../reference/ydb-cli/export-import/export-s3.md) | S3-совместимое хранилище | Нет | Аварийное восстановление, миграция, архивирование ||
+|| [Коллекции резервных копий](datamodel/backup-collection.md) | В кластере ([экспортируются](#backup-collections-storage) в S3 или файловую систему) | Да | Регулярные бэкапы больших производственных БД ||
```
```suggestion
|| [Копирование таблиц внутри кластера](#copy-table) | В кластере | Нет | Быстрая копия перед опасной операцией ||
|| [Дамп в файловую систему](#dump) | Файловая система | Нет | Разработка, тестирование, небольшие БД ||
|| [Экспорт в S3-совместимое хранилище](#s3) | S3-совместимое хранилище | Нет | Аварийное восстановление, миграция, архивирование ||
|| [Инкрементальное резервное копирование](#incremental-backup) | В кластере ([экспортируются](#backup-collections-storage) в S3 или файловую систему) | Да | Регулярные бэкапы больших производственных БД ||
```

CyberROFL commented on ydb/docs/ru/core/dev/resource-consumption-management.md:
Лишние изменения.

CyberROFL commented on ydb/docs/ru/core/concepts/datamodel/backup-collection.md:
```diff
@@ -0,0 +1,179 @@
+# Коллекция резервных копий
+
+Коллекция резервных копий организует полные и инкрементальные резервные копии выбранных [строковых таблиц](table.md#row-oriented-tables) в управляемые цепочки. Она позволяет **восстановить данные к состоянию на момент одной из резервных копий в цепочке** (полной или инкрементальной), обеспечивая защиту от случайной потери данных — ошибочных удалений или изменений. Произвольная «точка во времени»  ...
```
```suggestion
Коллекция резервных копий организует полные и инкрементальные резервные копии выбранных [строковых таблиц](table.md#row-oriented-tables) в управляемые цепочки. Она позволяет **восстановить данные к состоянию на момент одной из резервных копий в цепочке** (полной или инкрементальной), обеспечивая защиту от случайной потери данных — ошибочных удалений или изменений.
```

Я бы убрал это уточнение из введения.

CyberROFL commented on ydb/docs/ru/core/concepts/datamodel/backup-collection.md:
```diff
@@ -0,0 +1,179 @@
+
+- Быстрые операции резервного копирования и восстановления
+- Интегрированные механизмы безопасности
+- Не требуется внешняя инфраструктура
```
```suggestion
- Быстрые операции резервного копирования и восстановления.
- Интегрированные механизмы безопасности.
- Не требуется внешняя инфраструктура.
```

CyberROFL commented on ydb/docs/ru/core/concepts/datamodel/backup-collection.md:
```diff
@@ -0,0 +1,179 @@
+Для экспорта резервных копий во внешнее хранилище используйте {{ ydb-short-name }} CLI:
+
+- `ydb export s3` для S3-совместимого хранилища
+- `ydb tools dump` для файловой системы
```
```suggestion
- `ydb export s3` для S3-совместимого хранилища.
- `ydb tools dump` для файловой системы.
```

CyberROFL commented on ydb/docs/ru/core/recipes/backup-collections/multi-environment-setup.md:
```diff
@@ -0,0 +1,54 @@
+# Настройка резервного копирования для разных сред
+
+Настройка различных стратегий резервного копирования для сред разработки и продакшена.
```
Тут и далее:
```suggestion
Настройка различных стратегий резервного копирования для сред разработки и продуктовых сред.
```
Возможно, это минорное.

CyberROFL commented on ydb/docs/ru/core/dev/resource-consumption-management.md:
```diff
@@ -229,33 +229,6 @@ ydb -p <profile_name> sql -s 'select 1' --stats full --format json-unicode
 - `ResourcePoolId` — имя пула ресурсов, к которому был привязан запрос.
 - `QueuedTimeUs` — общее время ожидания запроса в очереди.
 
-### Статус выполняющегося запроса
```
Почему удалил?

Enjection replied:
Криво разрешил ребейз

CyberROFL commented on ydb/docs/ru/core/concepts/datamodel/backup-collection.md:
```diff
@@ -0,0 +1,179 @@
+Для экспорта резервных копий во внешнее хранилище используйте {{ ydb-short-name }} CLI:
+
+- `ydb export s3` для S3-совместимого хранилища.
+- `ydb tools dump` для файловой системы.
```
Не хватает ссылок.

CyberROFL commented on ydb/docs/ru/core/concepts/datamodel/backup-collection.md:
```diff
@@ -0,0 +1,179 @@
+
+### Фоновые операции
+
+Все операции резервного копирования и восстановления выполняются асинхронно, позволяя продолжать обычные операции с базой данных. Отслеживайте прогресс с помощью `ydb operation list incbackup`.
```
Не хватает ссылки.

Enjection committed:
Backup collections documentation

Add comprehensive documentation for backup collections feature:
- Concepts: backup.md, backup-collection.md (EN/RU)
- Schema object: datamodel/backup-collection.md (EN/RU)
- YQL reference: CREATE/BACKUP/RESTORE/DROP BACKUP COLLECTION (EN/RU)
- DevOps guide: backup-and-recovery.md (EN/RU)
- Recipes: getting-started, multi-environment, microservices, export/import, maintenance, validation (EN/RU)
- Glossary entries and TOC updates

Enjection committed:
Address docs review comments

- Clarify RESTORE restores to latest backup in chain only
- Note prefix-import workaround for earlier restore points
- Fix operation-list anchor {examples} -> {#examples}
- Fix EN --format description: Input -> Output format

Enjection committed:
Correct RESTORE conflict behavior in docs

RESTORE uses FailOnExist=true in schemeshard — it rejects the
operation if any target table already exists, rather than overwriting.
Updated both concept and YQL reference pages (RU/EN).

Enjection committed:
Align backup-collection intro with RESTORE semantics

Drops the "(full or incremental)" parenthetical per review feedback
and rewords to "latest backup in the chain" so the intro agrees with
the rest of the page and the RESTORE reference.

CyberROFL commented on ydb/docs/ru/core/concepts/backup.md:
```diff
@@ -0,0 +1,89 @@
+
+{% endnote %}
+
+## Коллекции резервных копий {#backup-collections}
+
+[Коллекция резервных копий](datamodel/backup-collection.md) — это механизм, который объединяет набор таблиц и управляет цепочкой их полных и инкрементальных резервных копий. Коллекции хранятся внутри кластера и могут быть [экспортированы во внешнее хранилище](datamodel/backup-collection.md#external-storage) для защиты от потери кластера.
+
+Подробнее об архитектуре, ограничениях, операциях и внутреннем устройстве коллекций см. в разделе [Коллекции резервных копий](datamodel/backup-collection.md).
```
А почему это осталось? Мне кажется, выше надо дать ссылку сразу на datamodel/backup-collection.md, а это убрать.

Enjection replied:
Т.е. вообще не описывать тут, просто 2 слова и ссылка?

CyberROFL replied:
Можешь еще чье-нибудь мнение узнать. Но мне кажется, это здесь это лишние детали, которые уже описаны в отдельной статье про коллекции.

CyberROFL commented on ydb/docs/ru/core/recipes/backup-collections/exporting-to-external-storage.md:
```diff
@@ -0,0 +1,45 @@
+
+## Экспорт в S3
+
+Для больших резервных копий или аварийного восстановления экспортируйте напрямую в S3-совместимое хранилище:
```
```suggestion
Для больших резервных копий или аварийного восстановления экспортируйте в S3-совместимое хранилище:
```
Иначе "напрямую" можно понять как "минуя локальное хранилище в кластере".

Enjection committed:
Update ydb/docs/ru/core/recipes/backup-collections/exporting-to-external-storage.md

Co-authored-by: Ilnaz Nizametdinov <i.nizametdinov@gmail.com>

Enjection committed:
Simplify backup collections section in backup.md to just link to dedicated article

CyberROFL commented on ydb/docs/ru/core/concepts/backup.md:
```diff
@@ -0,0 +1,87 @@
+
+Для восстановления необходима вся цепочка: сначала применяется полная копия, затем последовательно все инкременты. Восстановление выполняется к состоянию на момент последнего инкремента в цепочке.
+
+Инкрементальное резервное копирование реализуется с помощью [коллекций резервных копий](#backup-collections).
```
```suggestion
Инкрементальное резервное копирование реализуется с помощью [коллекций резервных копий](datamodel/backup-collection.md).
```

CyberROFL commented on ydb/docs/ru/core/concepts/backup.md:
```diff
@@ -0,0 +1,87 @@
+## Коллекции резервных копий {#backup-collections}
+
+Подробнее о коллекциях резервных копий см. в разделе [Коллекции резервных копий](datamodel/backup-collection.md).
+
```
Всё равно осталось...
```suggestion
```

CyberROFL committed:
Apply suggestions from code review

Co-authored-by: Ilnaz Nizametdinov <i.nizametdinov@gmail.com>

CyberROFL reviewed (commented):
Русская версия lgtm.

sintjuri commented:
на ревью у Андрея Фомичева

fomichev3000 commented on ydb/docs/ru/core/yql/reference/syntax/create-backup-collection.md:
````diff
@@ -0,0 +1,58 @@
+
+```yql
+CREATE BACKUP COLLECTION collection_name (
+    TABLE table_name [, TABLE another_table_name ...]
````
Я могу сделать Backup Collection для всей БД? Мне нужно

fomichev3000 commented on ydb/docs/ru/core/concepts/datamodel/backup-collection.md:
```diff
@@ -0,0 +1,179 @@
+Перед использованием коллекций резервных копий учтите эти ограничения:
+
+- **Только строковые таблицы**: [Колоночные таблицы](table.md#column-oriented-tables) не поддерживаются.
+- **Одна коллекция на таблицу**: Таблица может принадлежать только одной коллекции резервных копий. Чтобы включить таблицу в другую коллекцию, выполните [`DROP BACKUP COLLECTION`](../../yql/reference/syntax/drop-backup-collection.md) для текущей коллекции и создайте новую с нужным составом таблиц.
```
Ого! Почему?

sintjuri committed:
commit to rebuild

CyberROFL reviewed (commented):
Я уже выше [писал](https://github.com/ydb-platform/ydb/pull/23021#event-24792398853) lgtm.

sintjuri committed:
Update backup-collection.md
